### PR TITLE
Implemented sigmoid and log_sigmoid, Resolves #171

### LINKED ISSUE #171: Sigmoid function
### Feature description

Add the sigmoid activation function.

### PR DESCRIPTION
**Pull request by h4rr9**
implemented sigmoid and log_sigmoid, added tests to check for overflow

### CODE REVIEW

File: burn-tensor/src/tensor/activation/base.rs [hunk 1]
```diff
@@ -33,3 +33,20 @@ pub fn log_softmax<const D: usize, B: Backend>(tensor: Tensor<B, D>, dim: usize)
 
     tensor.sub(tensor_tmp)
 }
+
+/// Applies the softmax function.
```
Comment (nathanielsimard) [inline burn-tensor/src/tensor/activation/base.rs:37]: typo: `softmax` => `sigmoid`

File: burn-tensor/src/tensor/activation/base.rs [hunk 2]
```diff
@@ -33,3 +33,20 @@ pub fn log_softmax<const D: usize, B: Backend>(tensor: Tensor<B, D>, dim: usize)
...
+pub fn sigmoid<const D: usize, B: Backend>(tensor: Tensor<B, D>) -> Tensor<B, D> {
+    log_sigmoid(tensor).exp()
+}
+
+/// Applies the log softmax function.
```
Comment (nathanielsimard) [inline burn-tensor/src/tensor/activation/base.rs:42]: same typo

**Review by nathanielsimard — CHANGES_REQUESTED** [review]

File: burn-tensor/src/tensor/activation/base.rs (hunk 1 above)
Comment (h4rr9) [reply burn-tensor/src/tensor/activation/base.rs]: fixed the typo

File: burn-tensor/src/tensor/activation/base.rs (hunk 2 above)
Comment (h4rr9) [reply burn-tensor/src/tensor/activation/base.rs]: fixed the typo

**Review by nathanielsimard — APPROVED** [review]

**Pull request merged**